android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -1146,16 +1146,13 @@ private void setBackgroundImage(NTPImage ntpImage) {
         mBgImageView = (ImageView) findViewById(R.id.bg_image_view);
         mBgImageView.setScaleType(ImageView.ScaleType.MATRIX);
 
-        DisplayMetrics displayMetrics = new DisplayMetrics();
-        mActivity.getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
-        int mDeviceHeight = displayMetrics.heightPixels;
-        int mDeviceWidth = displayMetrics.widthPixels;
         ViewTreeObserver observer = mBgImageView.getViewTreeObserver();
         observer.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
             @Override
             public void onGlobalLayout() {
-                mWorkerTask = new FetchWallpaperWorkerTask(
-                        ntpImage, mDeviceWidth, mDeviceHeight, wallpaperRetrievedCallback);
+                mWorkerTask =
+                        new FetchWallpaperWorkerTask(ntpImage, mBgImageView.getMeasuredWidth(),
```

Comment:
@tapanmodh what's the reasoning here to use imageview size over the screen size ? is it related to top bar and bottom bar size ?

Reply:
yes, because of use of screen size, bottom parts of ntp image was getting cut. As per design instructions, we need to display full size of image's height in the ntp.